Rewrite multicluster installation docs (#8180)
istio/istio.io: content/en/docs/setup/install/multicluster/index.md

---
title: Multicluster Installation
description: Install an Istio mesh across multiple Kubernetes clusters.
weight: 30
aliases:
    - /docs/setup/kubernetes/multicluster-install/
    - /docs/setup/kubernetes/multicluster/
    - /docs/setup/kubernetes/install/multicluster/
keywords: [kubernetes,multicluster]

> test: yes

Reviewer: Although you do have test: set to yes which generates the snips.sh, and there are not test scripts created. The yes does show a checkmark on the page that this page is tested, when in fact there are no tests.

That said, I noticed that the scrips.sh did get lint tested to highlight problems in the snippets on the page.

Author: @frankbu added `yes` ... I assumed that was the goal, but not sure. I'm fine setting it back to `no`. I'd like to do the tests with a follow-on PR since it will be huge and complex.

Reviewer: OK. I'll let @frankbu respond if it should go back to no now that the scripts are tested and being there are no tests at the moment. As noted above, I can see a reason to have the scripts.sh created and linted when updating pages with snippets. But it also leaves a false impression that a page is tested.

Reviewer 2: As long as the test is coming in a followup PR, before the 1.8 release (which is our policy for new docs), I think it's OK if it's set to yes in preliminary.istio.io, so that (as @ericvn pointed out) we can at least be sure the commands have no lint errors.

owner: istio/wg-environments-maintainers
---
When installing Istio across multiple clusters, there are more things to
consider than with a [single cluster installation](/docs/setup/getting-started).

> Although there is no default installation procedure for a multicluster mesh, as it will

Reviewer 3: I'd rephrase:

This guide covers some of the most common concerns when creating a multi-cluster mesh using Istio:
...

I don't think we need the first part about things being different.

be different based on your cluster and network architecture, this guide covers some
of the most common concerns:

- [Network topologies](/docs/ops/deployment/deployment-models#network-models):
  one or two networks

- [Control plane topologies](/docs/ops/deployment/deployment-models#control-plane-models):
  multi-primary, primary-remote

Before you begin, review the [deployment models guide](/docs/ops/deployment/deployment-models)
which describes the foundational concepts used throughout this guide.

## Requirements

> This guide requires that you have two Kubernetes clusters with any of the

Reviewer 3: Maybe be specific that we mean kubernetes versions here? Unclear if this means you need two clusters with Istio installed.

[supported versions](/docs/setup/platform-setup).

If the clusters are on different networks (i.e. no direct connectivity between

> pods on different clusters), they must have ingress gateway services which are

Reviewer 3: Make it clear that those can be set up using Istio after Istio is installed.

Otherwise this reads like they need to set up ingress gateways before installing Istio.

Alternatively just drop this from here and talk about it when configuring the second cluster on a different network.

Author: I've removed this and added some basic text to the multicluster sections.

`CTX_BARNEY` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the `BARNEY` cluster.

For example:

{{< text bash >}}
$ export CTX_FRED=cluster-fred
$ export CTX_BARNEY=cluster-barney
{{< /text >}}

## Configure Trust

A multicluster service mesh deployment requires that you establish trust
between all clusters in the mesh. Depending on the requirements for your
system, there may be multiple options available for establishing trust.
See [certificate management](/docs/tasks/security/cert-management/) for
detailed descriptions and instructions for all available options.
Depending on which option you choose, the installation instructions for
Istio may change slightly.

This guide will assume that you use a common root to generate intermediate
certificates for each cluster. Follow the [instructions](/docs/tasks/security/cert-management/plugin-ca-cert/)

> to generate and push a secret to both the `FRED` and `BARNEY` clusters.

Reviewer 3: ```suggestion
to generate and push ca certificate secrets to both the `FRED` and `BARNEY` clusters.
```

{{< tip >}}
If you currently have a single cluster with a self-signed CA (as described
in [Getting Started](/docs/setup/getting-started/)), you’ll need to

> [change the CA](/docs/tasks/security/cert-management/) to support multiple

Reviewer 3: That link isn't clear, what do they actually need to do if they already have Istio installed with a self-signed cert? Can we point them at explicit instructions on how to migrate from self-signed to intermediate?

clusters and then reinstall Istio using the standard installation
instructions below.
{{< /tip >}}

## Install Istio

The steps for installing Istio on multiple clusters depend on your
requirements for network and control plane topology. This section

> illustrates the options for two clusters. Large systems may employ more

Reviewer 3: Meshes that span many clusters may employ more than one of these options.

than one of these options for meshes that span many clusters. See
[deployment models](/docs/ops/deployment/deployment-models) for more
information.

{{< tabset category-name="install-istio" >}}

{{< tab name="Multi-Primary" category-value="multi-primary" >}}

The steps that follow will install the Istio control plane on both `FRED` and
`BARNEY`, making each a {{< gloss >}}primary cluster{{< /gloss >}}. Both
clusters reside on the `EAST` network.

Each control plane observes the API Servers in both clusters for endpoints.
Service workloads communicate directly (pod-to-pod) across cluster boundaries.

{{< image width="75%"
    link="multi-primary.svg"
    caption="Multiple primary clusters on the same network"
    >}}

> ### Configure `FRED` as a primary

Reviewer 2: All of the section headings that are inside a `{{< tab ... >}}` don't work properly. Notice that if you click on any of these headings in the outline, it doesn't go to them. Not sure, what the best solution is, but one option is just to turn these headings into bold text.
```suggestion
**Configure `FRED` as a primary**
```

Author: @frankbu if we make this change they simply won't appear in the outline, right? I'm not sure if that's much better. From a readability standpoint, having them as headings (rather than just bold) seems more clear.

Install Istio on `FRED`:

{{< text bash >}}

> $ istioctl --context=${CTX_FRED} -f - <<EOF

Reviewer 3: Super minor nit: I'd rather show us putting that operator config into a file (echo to a file for example) and then passing that file to istioctl install, rather than inlining it. Sets them up better for editing that file as they desire.

Also you missed the "install" part of the command.

apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: BEDROCK
      multiCluster:
        clusterName: FRED
      network: EAST
      meshNetworks:
        EAST:
          endpoints:
          - fromRegistry: FRED
          - fromRegistry: BARNEY
          gateways:

>           - registryServiceName: istio-ingressgateway.istio-system.svc.cluster.local

Reviewer 4: ```suggestion
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
```

            port: 15443
EOF
{{< /text >}}

### Configure `BARNEY` as a primary

Install Istio on `BARNEY`:

{{< text bash >}}
$ istioctl --context=${CTX_BARNEY} -f - <<EOF
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: BEDROCK
      multiCluster:
        clusterName: BARNEY
      network: EAST
      meshNetworks:
        EAST:
          endpoints:
          - fromRegistry: BARNEY
          - fromRegistry: FRED
          gateways:

>           - registryServiceName: istio-ingressgateway.istio-system.svc.cluster.local

Reviewer 4: ```suggestion
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
```

            port: 15443
EOF
{{< /text >}}

### Enable Endpoint Discovery

Install a remote secret in `BARNEY` that provides access to `FRED`’s API server.

{{< text bash >}}
$ istioctl x create-remote-secret \
    --context=${CTX_FRED} \
    --name=FRED | \
    kubectl apply -f - --context=${CTX_BARNEY}
{{< /text >}}

Install a remote secret in `FRED` that provides access to `BARNEY`’s API server.

{{< text bash >}}
$ istioctl x create-remote-secret \
    --context=${CTX_BARNEY} \
    --name=BARNEY | \
    kubectl apply -f - --context=${CTX_FRED}
{{< /text >}}

{{< /tab >}}

{{< tab name="Multi-Primary, Multi-Network" category-value="multi-primary-multi-network" >}}

The following steps will install the Istio control plane on both `FRED` and
`BARNEY`, making each a {{< gloss >}}primary cluster{{< /gloss >}}. `FRED` is
on the `WEST` network, while `BARNEY` is on the `EAST` network.

> Both `FRED` and `BARNEY` observe the API Servers in each cluster for endpoints.

Reviewer 3: I think this image is wrong, it has istiod directly observing the api server in the other cluster, but if it is on a different network it can't do that right? Is istiod using the E-W gateway to get to the api server in the other cluster? Or is the api server exposed some other way? Maybe we should mention how that part works and any requirements on istiod having access to the api server for other clusters?

Reviewer 3: I think many api servers from cloud providers are publicly accessible through an NLB but we should be explicit about that requirement. Some don't meet that and API servers are private, which means this may not work. And for on-prem Kubernetes installs all bets are off...

Author: Yeah that was the assumption .. that API Servers are publicly available. I'll add that to the requirements at the top.

    >}}

### Configure `FRED` as a primary with services exposed

Install Istio on `FRED`:

{{< text bash >}}
$ istioctl --context=${CTX_FRED} -f - <<EOF
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: BEDROCK
      multiCluster:
        clusterName: FRED
      network: WEST
      meshNetworks:
        WEST:
          endpoints:
          - fromRegistry: FRED
          gateways:
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
            port: 15443
        EAST:
          endpoints:
          - fromRegistry: BARNEY
          gateways:
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
            port: 15443
EOF
{{< /text >}}

Install a gateway in `FRED` that is dedicated to
[east-west](https://en.wikipedia.org/wiki/East-west_traffic) traffic. By
default, this gateway will be public on the Internet. Production systems may
require additional access restrictions (e.g. via firewall rules) to prevent
external attacks. Check with your cloud vendor to see what options are
available.

{{< text bash >}}
$ CLUSTER=FRED NETWORK=WEST \

>     samples/multicluster/gen-eastwest-gateway.sh | \

Reviewer 3: Same here, I'd prefer putting in a file and then having them kubectl apply it so they know they can save that file into their source control system.

    samples/multicluster/expose-services.yaml
{{< /text >}}

### Enable Endpoint Discovery for `FRED` and `BARNEY`

Install a remote secret in `BARNEY` that provides access to `FRED`’s API server.

{{< text bash >}}
$ istioctl x create-remote-secret \
  --context=${CTX_FRED} \
  --name=FRED | \
  kubectl apply -f - --context=${CTX_BARNEY}
{{< /text >}}

Install a remote secret in `FRED` that provides access to `BARNEY`’s API server.

{{< text bash >}}
$ istioctl x create-remote-secret \
  --context=${CTX_BARNEY} \
  --name=BARNEY | \
  kubectl apply -f - --context=${CTX_FRED}
{{< /text >}}

{{< /tab >}}

{{< tab name="Primary-Remote" category-value="primary-remote" >}}

The following steps will install the Istio control plane on `FRED` (the
{{< gloss >}}primary cluster{{< /gloss >}}) and configure `BARNEY` (the
{{< gloss >}}remote cluster{{< /gloss >}}) to use the control plane in `FRED`. Both
clusters reside on the `EAST` network.

`FRED` will be configured to observe the API Servers in both clusters for
endpoints. In this way, the control plane will be able to provide service
discovery for workloads in both clusters. Service workloads across cluster

> boundaries communicate indirectly, via dedicated

Reviewer 3: This is wrong, they're going direct, it is only the istiod connection that goes through a gateway.

We should talk here about how control plane traffic goes through a gateway but data plane traffic goes direct. The control plane is indirect because of bootstrapping, we need a stable IP address for istiod and we use an ingress gateway to get that.

[east-west](https://en.wikipedia.org/wiki/East-west_traffic) gateways.

{{< image width="75%"
    link="primary-remote.svg"
    caption="Primary and remote clusters on the same network"
    >}}

### Configure `FRED` as a primary with control plane exposed

Install Istio on `FRED`:

{{< text bash >}}
$ istioctl --context=${CTX_FRED} -f - <<EOF
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: BEDROCK
      multiCluster:
        clusterName: FRED
      network: EAST
      meshNetworks:
        ${NETWORK1}:
          endpoints:
          - fromRegistry: FRED
          - fromRegistry: BARNEY
          gateways:
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
            port: 15443
EOF
{{< /text >}}

Install a gateway in the `FRED` that is dedicated to
[east-west](https://en.wikipedia.org/wiki/East-west_traffic) traffic. By
default, this gateway will be public on the Internet. Production systems may
require additional access restrictions (e.g. via firewall rules) to prevent
external attacks. Check with your cloud vendor to see what options are
available.

{{< text bash >}}

> $ CLUSTER=FRED NETWORK=WEST \

Reviewer 3: We don't need the E-W gateway (there is no WEST anyway).

Author: We're using the term E-W here to describe the traffic direction, it's not referring to the network names.

It is a bit confusing since we named our networks EAST and WEST. I guess we could rename the gateway to something like `in-mesh` or `inter-cluster`, but `east-west` is a fairly common term for this. Perhaps we rename the networks in these examples?

apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: BEDROCK
      multiCluster:
        clusterName: BARNEY
      network: EAST
      remotePilotAddress: ${DISCOVERY_ADDRESS}
EOF
{{< /text >}}

### Enable Endpoint Discovery for `BARNEY`

Create a remote secret that will allow the control plane in `FRED` to access the
API Server in `BARNEY` for endpoints.

{{< text bash >}}
$ istioctl x create-remote-secret \
    --context=${CTX_BARNEY} \
    --name=BARNEY | \
    kubectl apply -f - --context=${CTX_FRED}
{{< /text >}}

{{< /tab >}}

{{< tab name="Primary-Remote, Multi-Network" category-value="primary-remote-multi-network" >}}

The following steps will install the Istio control plane on `FRED` (the
{{< gloss >}}primary cluster{{< /gloss >}}) and configure `BARNEY` (the
{{< gloss >}}remote cluster{{< /gloss >}}) to use the control plane in `FRED`.
`FRED` is on the `WEST` network, while `BARNEY` is on the `EAST` network.

`FRED` will be configured to observe the API Servers in both clusters for
endpoints. In this way, the control plane will be able to provide service
discovery for workloads in both clusters.

> Services in the `BARNEY` will reach the control plane in the `FRED` via a dedicated

Reviewer 3: ```suggestion
Services in the `BARNEY` cluster will reach the control plane in the `FRED` cluster via a dedicated
```

request, `HelloWorld` will include its version in its response.

We will also deploy the `Sleep` container to both clusters. We will use these
pods as the source of requests to the `HelloWorld` service,
simulating in-mesh traffic. Finally, after generating traffic, we will observe
which cluster received the requests.

### Deploy the `HelloWorld` Service

In order to make the `HelloWorld` service callable from any cluster, the DNS
lookup must succeed in each cluster (see
[deployment models](/docs/ops/deployment/deployment-models#dns-with-multiple-clusters)
for details). We will address this by deploying the `HelloWorld` Service to
each cluster in the mesh.

To begin, create the `sample` namespace in each cluster:

{{< text bash >}}
$ kubectl create --context=${CTX_FRED} namespace sample
$ kubectl create --context=${CTX_BARNEY} namespace sample
{{< /text >}}

Enable automatic sidecar injection for the `sample` namespace:

{{< text bash >}}
$ kubectl label --context=${CTX_FRED} namespace sample \
    istio-injection=enabled
$ kubectl label --context=${CTX_BARNEY} namespace sample \
    istio-injection=enabled
{{< /text >}}

> Deploy the `HelloWorld` service to both clusters:

Reviewer 3: ```suggestion
Create the `HelloWorld` service in both clusters:
```

{{< /text >}}

## Configure Trust

A multicluster service mesh deployment requires that you establish trust
between all clusters in the mesh. Depending on the requirements for your
system, there may be multiple options available for establishing trust.
See [certificate management](/docs/tasks/security/cert-management/) for
detailed descriptions and instructions for all available options.
Depending on which option you choose, the installation instructions for
Istio may change slightly.

This guide will assume that you use a common root to generate intermediate
certificates for each cluster. Follow the [instructions](/docs/tasks/security/cert-management/plugin-ca-cert/)
to generate and push a ca certificate secrets to both the `CLUSTER1` and `CLUSTER2`
clusters.

{{< tip >}}
If you currently have a single cluster with a self-signed CA (as described
in [Getting Started](/docs/setup/getting-started/)), you need to
change the CA using one of the methods described in
[certificate management](/docs/tasks/security/cert-management/). Changing the
CA typically requires reinstalling Istio. The installation instructions
below may have to be altered based on your choice of CA.

> {{< /tip >}}

Reviewer 5: Could we have some clarification around `Changing the
CA typically requires reinstalling Istio`?  Do we mean just reinstall the control plane?  Is restarting data plane needed or can Istio auto distribute the new key/certs to each sidecar?

Reviewer 5: cc @myidpt @howardjohn

## Install Istio

The steps for installing Istio on multiple clusters depend on your
requirements for network and control plane topology. This section
illustrates the options for two clusters. Meshes that span many clusters may
employ more than one of these options. See
[deployment models](/docs/ops/deployment/deployment-models) for more
information.

{{< tabset category-name="install-istio" >}}

{{< tab name="Multi-Primary" category-value="multi-primary" >}}

The steps that follow will install the Istio control plane on both `CLUSTER1` and
`CLUSTER2`, making each a {{< gloss >}}primary cluster{{< /gloss >}}. Both
clusters reside on the `NETWORK1` network, meaning there is direct
connectivity between the pods in both clusters.

Each control plane observes the API Servers in both clusters for endpoints.

Service workloads communicate directly (pod-to-pod) across cluster boundaries.

{{< image width="75%"
    link="multi-primary.svg"
    caption="Multiple primary clusters on the same network"
    >}}

> **Configure `CLUSTER1` as a primary**

Reviewer 2: Maybe to get around the lint error you can make these dummy headings list items for now:
```suggestion
- **Configure `CLUSTER1` as a primary**
```
We can revisit the layout and improve in a followup PR.

Reviewer 2: Maybe something like this is another option? https://stackoverflow.com/questions/52062672/rmarkdown-add-header-under-tabset

> - **Configure `CLUSTER1` as a primary**

Reviewer 2: This doesn't look so good in the preview. Have you tried using html tags?
```suggestion
<h3>Configure `CLUSTER1` as a primary</h3>
```
If that doesn't work either, then maybe we should merge it as is and investigate other ideas in a followup PR.

$ cat <<EOF > ./cluster1.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: MESH1
      multiCluster:
        clusterName: CLUSTER1
      network: NETWORK1
      meshNetworks:
        NETWORK1:
          endpoints:
          - fromRegistry: CLUSTER1
          - fromRegistry: CLUSTER2
          gateways:
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
            port: 15443
EOF
$ istioctl install --context=${CTX_CLUSTER1} -f cluster1.yaml
{{< /text >}}

**Configure `CLUSTER2` as a primary**

Install Istio on `CLUSTER2`:

{{< text bash >}}
$ cat <<EOF > ./cluster2.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: MESH1
      multiCluster:
        clusterName: CLUSTER2
      network: NETWORK1
      meshNetworks:
        NETWORK1:
          endpoints:
          - fromRegistry: CLUSTER2
          - fromRegistry: CLUSTER1
          gateways:
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
            port: 15443
EOF
$ istioctl install --context=${CTX_CLUSTER2} -f cluster2.yaml
{{< /text >}}

> 

Reviewer 5: For multi primary, the diagram didn't show gateways while the plan showed gateways.  This is a bit confusing when I reviewed this tab.

Author: Are you referring to the presence of `istio-eastwestgateway` in the `IstioOperator` yaml? Thing is that the gateway isn't needed for this case, so we don't actually deploy it. Not sure if it's valid to leave out `gateways` altogether, but if that works maybe that's a better solution here.

@stevenctl WDYT?

Reviewer 5: I assume the gateways aren't used for workloads to communicate if so, it is confusing we also call it eastwestgateway.

Reviewer 4: I do agree that we don't need to show the gateways in the multi-primary guide, they're not really necessary. 

@linsun 
The eastwest gateways are used for:
- multinetwork workload traffic
- discovery from outside the clsuter

eastwest sort of implies it's mostly used for workload traffic, so I can see how that name is confusing.

Author: @linsun east-west is a pretty standard term, but I'm open to other names if they're more clear.

Reviewer 5: If removing the gateways are okay for this example, I'd vote for removing it because it is quite confusing to have it :) @stevenctl @nmittler

Reviewer 5: @nmittler the name east-west is good.  It is just confusing here when it is not used for workload traffic.

            port: 15443
        NETWORK2:
          endpoints:
          - fromRegistry: CLUSTER2
          gateways:
          - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local
            port: 15443
EOF
$ istioctl install --context=${CTX_CLUSTER1} -f cluster1.yaml
{{< /text >}}

Install a gateway in `CLUSTER1` that is dedicated to
[east-west](https://en.wikipedia.org/wiki/East-west_traffic) traffic. By
default, this gateway will be public on the Internet. Production systems may
require additional access restrictions (e.g. via firewall rules) to prevent
external attacks. Check with your cloud vendor to see what options are
available.

{{< text bash >}}
$ CLUSTER=CLUSTER1 NETWORK=NETWORK1 \
    samples/multicluster/gen-eastwest-gateway.sh | \
    kubectl apply --context=${CTX_CLUSTER1} -f -
{{< /text >}}

Since the clusters are on separate networks, we need to expose all services
(*.local) on the east-west gateway in both clusters. While this gateway is
public on the Internet, services behind it can only be accessed by services
with a trusted mTLS certificate and workload ID, just as if they were on the

> same network.

Reviewer 5: Ok this is where I think there is some gap when comparing with our previous replicated control plane model.   The doc here instructs users to expose all services while the replicated control plane instructs users to selectively expose services.   I don't think we have that function ready yet as we are waiting on k8s for the multicluster service API.

This is also where I'm not sure if we are ready to remove the replicated control plane doc yet without an equivalent replacement.  

@nmittler @stevenctl @rshriram @frankbu

Author: I'm not sure there is a gap, really. This should be the default behavior for a service mesh that spans multiple clusters. The updated [Deployment Models](https://preliminary.istio.io/latest/docs/ops/deployment/deployment-models/#dns-with-multiple-clusters) doc discusses selective service exposure via `Service`/`ServiceEntry`, but that is beyond the scope of this guide.

Reviewer 5: Are you saying it is exposed to the other cluster by default but what services to import are controlled by creating service entry or the corresponding service in the consuming cluster?

Author: You can further clamp it down by being more selective when we expose services on the gateway. I can add a note in the text if that helps?

Both clusters reside on the `NETWORK1` network, meaning there is direct
connectivity between the pods in both clusters.

`CLUSTER1` will be configured to observe the API Servers in both clusters for
endpoints. In this way, the control plane will be able to provide service
discovery for workloads in both clusters.

Service workloads communicate directly (pod-to-pod) across cluster boundaries.

Services in `CLUSTER2` will reach the control plane in `CLUSTER1` via a
dedicated gateway for [east-west](https://en.wikipedia.org/wiki/East-west_traffic)
traffic.

{{< image width="75%"
    link="primary-remote.svg"
    caption="Primary and remote clusters on the same network"
    >}}

- **Configure `CLUSTER1` as a primary with control plane exposed**

Install Istio on `CLUSTER1`:

{{< text bash >}}
$ cat <<EOF > ./cluster1.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      meshID: MESH1
      multiCluster:
        clusterName: CLUSTER1
      network: NETWORK1
      meshNetworks:
        ${NETWORK1}:
          endpoints:
          - fromRegistry: CLUSTER1
          - fromRegistry: CLUSTER2
          gateways:

>           - registryServiceName: istio-eastwestgateway.istio-system.svc.cluster.local

Reviewer 5: Similar comment as earlier - Is gateway required here?

Reviewer 5: Ok i think it is required here for workloads on remote cluster to reach out to istiod... but not for workloads from remote cluster to reach out to cluster 1.

Author: yup, exactly.
